atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/QosCassandraReadTestSuite.java

```diff
@@ -0,0 +1,287 @@
+import com.palantir.atlasdb.config.ImmutableSweepConfig;
+import com.palantir.atlasdb.factory.TransactionManagers;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.qos.config.ImmutableQosClientConfig;
+import com.palantir.atlasdb.qos.config.ImmutableQosLimitsConfig;
+import com.palantir.atlasdb.qos.ratelimit.RateLimitExceededException;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoSchema;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManager;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.common.base.BatchingVisitable;
+import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.configuration.ShutdownStrategy;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+import com.palantir.docker.compose.logging.LogDirectory;
+import com.palantir.remoting.api.config.service.HumanReadableDuration;
+
+public class QosCassandraReadTestSuite {
+    private static final Random random = new Random();
+    private static final int CASSANDRA_PORT_NUMBER = 9160;
+    private static SerializableTransactionManager serializableTransactionManager;
+    private static final int readBytesPerSecond = 10_000;
+    private static final int writeBytesPerSecond = 10_000;
+    private static final int ONE_TODO_SIZE_IN_BYTES = 1050;
+
+    @ClassRule
+    public static DockerComposeRule docker = DockerComposeRule.builder()
+            .file("src/test/resources/cassandra-docker-compose.yml")
+            .waitingForService("cassandra", Container::areAllPortsOpen)
+            .saveLogsTo(LogDirectory.circleAwareLogDirectory(QosCassandraReadTestSuite.class))
+            .shutdownStrategy(ShutdownStrategy.AGGRESSIVE_WITH_NETWORK_CLEANUP)
+            .build();
+
+    @BeforeClass
+    public static void createTransactionManagerAndWriteData() {
+        serializableTransactionManager = TransactionManagers.builder()
+                .config(getAtlasDbConfig())
+                .runtimeConfigSupplier(QosCassandraReadTestSuite::getAtlasDbRuntimeConfig)
+                .schemas(ImmutableList.of(TodoSchema.getSchema()))
+                .userAgent("qos-test")
+                .buildSerializable();
+
+        Awaitility.await()
+                .atMost(Duration.ONE_MINUTE)
+                .pollInterval(Duration.ONE_SECOND)
+                .until(serializableTransactionManager::isInitialized);
+
+        serializableTransactionManager
+                .runTaskWithRetry((transaction) -> {
+                    writeNTodosOfSize(transaction, 200, 1_000);
```

Comment:
nit, `ONE_TODO_SIZE_IN_BYTES` is derived from this (plus a small constant overhead)